tests/unittests/tests-priority_pktqueue/tests-priority_pktqueue.c

```diff
@@ -0,0 +1,290 @@
+
+    gnrc_priority_pktqueue_push(&pkt_queue, &elem1);
+    head = gnrc_priority_pktqueue_head(&pkt_queue);
+    TEST_ASSERT(head == &pkt1);
+    TEST_ASSERT_EQUAL_INT(1, head->users);
+    TEST_ASSERT_EQUAL_INT(1, ((gnrc_priority_pktqueue_node_t *)(pkt_queue.first))->priority);
+    TEST_ASSERT_NULL(head->next);
+    TEST_ASSERT_EQUAL_STRING(TEST_STRING8, head->data);
+    TEST_ASSERT_EQUAL_INT(sizeof(TEST_STRING8), head->size);
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF, head->type);
+    TEST_ASSERT_EQUAL_INT(1, gnrc_priority_pktqueue_length(&pkt_queue));
+
+    gnrc_priority_pktqueue_push(&pkt_queue, &elem2);
+    head = gnrc_priority_pktqueue_head(&pkt_queue);
+
+    TEST_ASSERT(head == &pkt1);
+    TEST_ASSERT_EQUAL_INT(1, head->users);
+    TEST_ASSERT_EQUAL_INT(1, ((gnrc_priority_pktqueue_node_t *)(pkt_queue.first))->priority);
+    TEST_ASSERT_NULL(head->next);
+    TEST_ASSERT_EQUAL_STRING(TEST_STRING8, head->data);
+    TEST_ASSERT_EQUAL_INT(sizeof(TEST_STRING8), head->size);
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF, head->type);
+    TEST_ASSERT_EQUAL_INT(2, gnrc_priority_pktqueue_length(&pkt_queue));
+
+    gnrc_priority_pktqueue_push(&pkt_queue, &elem3);
+    head = gnrc_priority_pktqueue_head(&pkt_queue);
+
+    TEST_ASSERT(head == &pkt3);
+    TEST_ASSERT_EQUAL_INT(1, head->users);
+    TEST_ASSERT_EQUAL_INT(0, ((gnrc_priority_pktqueue_node_t *)(pkt_queue.first))->priority);
+    TEST_ASSERT_NULL(head->next);
+    TEST_ASSERT_EQUAL_STRING(TEST_STRING16, head->data);
+    TEST_ASSERT_EQUAL_INT(sizeof(TEST_STRING16), head->size);
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_UNDEF, head->type);
+    TEST_ASSERT_EQUAL_INT(3, gnrc_priority_pktqueue_length(&pkt_queue));
+
+    gnrc_priority_pktqueue_flush(&pkt_queue);
+
+    head = gnrc_priority_pktqueue_head(&pkt_queue);
+    TEST_ASSERT_NULL(head);
+    TEST_ASSERT_NULL(pkt_queue.first);
+    TEST_ASSERT_EQUAL_INT(0, gnrc_priority_pktqueue_length(&pkt_queue));
+}
+
+static void test_gnrc_priority_pktqueue_pop_empty(void)
+{
+    gnrc_pktsnip_t *res;
+
+    res = gnrc_priority_pktqueue_pop(&pkt_queue);
+
+    TEST_ASSERT_NULL(pkt_queue.first);
+    TEST_ASSERT_NULL(res);
+    TEST_ASSERT_EQUAL_INT(0, gnrc_priority_pktqueue_length(&pkt_queue));
+}
+
+
+static void test_gnrc_priority_pktqueue_pop(void)
+{
+    gnrc_pktsnip_t pkt1 = PKT_INIT_ELEM_STATIC_DATA(TEST_STRING8, NULL);
+    gnrc_pktsnip_t pkt2 = PKT_INIT_ELEM_STATIC_DATA(TEST_STRING16, NULL);
+    gnrc_priority_pktqueue_node_t elem1 = PRIORITY_PKTQUEUE_NODE_INIT(1,&pkt1);
+    gnrc_priority_pktqueue_node_t elem2 = PRIORITY_PKTQUEUE_NODE_INIT(0,&pkt2);
+    gnrc_pktsnip_t *res;
+    gnrc_pktsnip_t *head;
+
+    gnrc_priority_pktqueue_push(&pkt_queue, &elem1);
+    gnrc_priority_pktqueue_push(&pkt_queue, &elem2);
+
+    res = gnrc_priority_pktqueue_pop(&pkt_queue);
+
+    TEST_ASSERT(res == &pkt2);
+    TEST_ASSERT_EQUAL_INT(0,((priority_queue_node_t *)&elem2)->data);
```

Comment:
I did this assert since, inside `gnrc_priority_pktqueue_pop`, it will call `_free_node()` in which `priority_queue_node_init()` will be further called to initialize the popped `elem`. So I want to check the result of initialization of `elem`. I checked that  `priority_queue_node_init()` will set the `data` to `0`, that's why I did this kind of assert.. 

But, you are right, I should do this more concisely by `TEST_ASSERT_NULL(elem2.pkt);`